IOException on DockerBuildImage Task

Running the following DockerBuildImage Task throws an IOException:

`task createDockerfile(type: Dockerfile) {
    destFile = project.file('build/Dockerfile')
    from 'ubuntu:12.04'
}

task buildDockerImage(type: DockerBuildImage) {
    dependsOn createDockerfile
    inputDir = createDockerfile.destFile.get().asFile.parentFile
    tag = 'bmuschko/myimage:latest'

}`

Stacktrace:
`org.gradle.api.UncheckedIOException: java.io.IOException: Die Syntax f├╝r den Dateinamen, Verzeichnisnamen oder die Datentr├ñgerbezeichnung ist falsch
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.canonicalizedPaths(DefaultTaskExecutionPlan.java:686)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.getOutputPaths(DefaultTaskExecutionPlan.java:770)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.resolveMutations(DefaultTaskExecutionPlan.java:611)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.getResolvedMutationInfo(DefaultTaskExecutionPlan.java:597)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.selectNext(DefaultTaskExecutionPlan.java:552)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$2.transform(DefaultTaskPlanExecutor.java:174)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker$2.transform(DefaultTaskPlanExecutor.java:161)
        at org.gradle.internal.resources.DefaultResourceLockCoordinationService.withStateLock(DefaultResourceLockCoordinationService.java:45)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.executeWithWork(DefaultTaskPlanExecutor.java:161)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$ExecutorWorker.run(DefaultTaskPlanExecutor.java:130)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.io.IOException: Die Syntax f├╝r den Dateinamen, Verzeichnisnamen oder die Datentr├ñgerbezeichnung ist falsch
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.canonicalizedPaths(DefaultTaskExecutionPlan.java:681)
        ... 12 more`


I had to Debug Gradle to see whats going on here.
The exception happens in 
DefaultTaskExecutionPlan:681

Gradle tries to resolve the following path:
`F:\*******\build\.docker\:buildDockerImage-imageId.txt`
That fails with an IOException because of the ":"

### Your Environment
Gradle 4.10.2
gradle-docker-plugin 4.0.1
Windows

--- a/src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy
+++ b/src/functTest/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImageFunctionalTest.groovy
@@ -132,7 +132,7 @@ class DockerBuildImageFunctionalTest extends AbstractGroovyDslFunctionalTest {
                 }
             }
         """
-        File imageIdFile = new File(projectDir, 'build/.docker/:buildImage-imageId.txt')
+        File imageIdFile = new File(projectDir, 'build/.docker/buildImage-imageId.txt')
 
         when:
         BuildResult result = build('verifyImageId')

--- a/src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy
+++ b/src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerBuildImage.groovy
@@ -108,7 +108,8 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
     DockerRegistryCredentials registryCredentials
 
     /**
-     * Output file containing the image ID of the built image. Defaults to "$buildDir/.docker/$taskpath-imageId.txt".
+     * Output file containing the image ID of the built image. Defaults to "$buildDir/.docker/$taskpath-imageId.txt"
+     * with the leading colon of taskpath stripped and remaining colons replaced with underscores.
      *
      * @since 4.0.0
      */
@@ -121,7 +122,9 @@ class DockerBuildImage extends AbstractDockerRemoteApiTask implements RegistryCr
 
     DockerBuildImage() {
         inputDir.set(project.file('docker'))
-        File imageIdTextFile = new File(project.buildDir, ".docker/${path}-imageId.txt")
+
+        String safeTaskPath = path.replaceFirst("^:", "").replaceAll(":", "_")
+        File imageIdTextFile = new File(project.buildDir, ".docker/${safeTaskPath}-imageId.txt")
         imageIdFile.set(imageIdTextFile)
 
         if (imageIdTextFile.isFile()) {

